lib/services/mfa.go

```diff
@@ -0,0 +1,38 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package services
+
+import (
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+// NewExpiredReusableMFAResponseError returns an access-denied error hinting MFA
+// validation failure caused by expired reusable MFA response.
+func NewExpiredReusableMFAResponseError(errMsg string) error {
+	return trace.AccessDenied("Reusable MFA response validation failed and possibly expired: %s", errMsg)
+}
+
+// IsExpiredReusableMFAResponseError checks if received error is hinting expired
+// reusable MFA response.
+func IsExpiredReusableMFAResponseError(err error) bool {
+	return trace.IsAccessDenied(err) &&
+		strings.Contains(err.Error(), NewExpiredReusableMFAResponseError("").Error())
+}
```

Comment:
You should be able to actually make an error variable so that you can check it with `error.Is` instead of a weaker string comparison. If my suggestion below doesn't work exactly, checkout `ErrAdminActionMFARequired` for a functional example.

```suggestion
var ErrExpiredReusableMFAResponse = trace.AccessDeniedError{Message: "Reusable MFA response validation failed and possibly expired"}

// NewExpiredReusableMFAResponseError returns an access-denied error hinting MFA
// validation failure caused by expired reusable MFA response.
func NewExpiredReusableMFAResponseError(errMsg string) error {
	return trace.Wrap(ErrorExpiredReusableMFAResponse, errMsg)
}

// IsExpiredReusableMFAResponseError checks if received error is hinting expired
// reusable MFA response.
func IsExpiredReusableMFAResponseError(err error) bool {
	return errors.Is(ErrorExpiredReusableMFAResponse)
}
```

Reply:
I tried it but it doesn't work when going through grpc, where the two messages get merged into one.

so either i keep the string comparison here, or i dropped original message like `trace.Wrap(&mfa.ExpiredReusableMFAResponse /* do not pass the original error*/)`

what's your thought on this

Reply:
I see, I guess this is the same issue with `NewPrivateKeyPolicyError` then. However, I think the original error might not be important context for the user? It should look something like `webauthn data <backend-key> not found`, right, right? Up to you whether you think the original error is important or can be dropped.

Reply:
done [298ffe1](https://github.com/gravitational/teleport/pull/54069/commits/298ffe1429b141fbc2671b70502f5f6be5799d57)